src/openfermion/utils/_trotter_qasm_conversion.py

```diff
@@ -0,0 +1,331 @@
+
+'''
+Description:
+		Approximates the exponential of an operator composed of Pauli
+		matrices, by Trotterization. Outputs QASM format to a python
+		stream.
+
+		Change-of-basis gates:
+		H			: Z to X basis (and back)
+		Rx(pi/2)	: Z to Y basis ..... 
+		Rx(-pi/2) 	: Y to Z basis ..... 
+
+'''
+
+
+
+
+def third_order_trotter_helper(hamiltonian_dict,trotterized_list,op_a, op_b):
+	'''
+	This function recursively trotterizes a QubitOperator according
+		to the scheme:
+		e^(A+B) = e^(7/24 * A) e^(2/3 * B) e^(3/4 * A) e^(-1/24 * A) e^(B)
+
+	Used in trotterize() function.
+
+	Params: 
+		trotterized_list: list that the function populates with trotterized
+				QubitOperators
+		op_a: first term in hamiltonian as a QubitOperator
+		op_b: the rest of the terms in the hamiltonian as a list of
+				QubitOperator term dictionary keys
+
+	Returns: None
+	'''
+	trotterized_list.append(7.0/24.0 * op_a)
+	if len(op_b) == 1:
+		trotterized_list.append(2.0 / 3.0 * QubitOperator(op_b[0],
+			hamiltonian_dict[op_b[0]]))
+	else:
+		btemp = op_b[1:]
+		for i in btemp:
+			hamiltonian_dict[i] *= 2.0 / 3.0
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		atemp *= 2.0 / 3.0
+		third_order_trotter_helper(trotterized_list, atemp, btemp)
+	trotterized_list.append(3.0/4.0 * op_a)
+
+	if op_b.terms.size == 1:
+		trotterized_list.append(-2.0/3.0 * QubitOperator(op_b[0],
+			hamiltonian_dict[op_b[0]]))
+	else:
+		btemp = op_b[1:]
+		for i in btemp:
+			hamiltonian_dict[i] *= -2.0 / 3.0
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		atemp *= -2.0 / 3.0
+		third_order_trotter_helper(trotterized_list,atemp,btemp)
+	trotterized_list.append(-1.0 / 24.0 * op_a)
+
+	if op_b.terms.size == 1:
+		trotterized_list.append(QubitOperator(op_b[0],
+			hamiltonian_dict[op_b[0]]))
+	else:
+		btemp = op_b[1:]
+		atemp = QubitOperator(op_b[0],hamiltonian_dict[op_b[0]])
+		third_order_trotter_helper(trotterized_list, atemp, btemp)
+
+
+
+
+def trotterize(
+	hamiltonian,trotter_steps = 1, 
+	trotter_order = 1, 
+	term_ordering = [], 
+	k_exp = 1.):
+	'''
+	This function trotterizes a hamiltonian to an order 1-3 and 
+	for a given number of trotter steps
+
+	Params:
+		hamiltonian: full hamiltonian as a QubitOperator.
+		trotter_steps: number of trotter steps as an integer.
+		order: order of trotterization as an integer from 1-3.
+		term_ordering (optional): list of tuples (QubitOperator terms
+				dictionary keys) that specifies order of terms when
+				trotterizing.
+		k_exp (optional): exponential factor to all terms when trotterizing.
+	
+	Returns:
+		tuple: (A list of single Pauli-term QubitOperators, the number of 
+		qubits required by the hamiltonian as an integer)
+	
+	Raises:
+		ValueError if order > 3 or order <= 0 or trotter_steps > 100,
+			TypeError for incorrect type 
+	'''
+	
+
+	if trotter_order > 3 or trotter_order <= 0:
+		raise ValueError("Invalid trotter order: " + str(order))
+	elif trotter_steps <= 0 or trotter_steps > 100: 
```

Comment:
Why are trotter steps capped to 100? If this is a hard requirement for some reason, it should be listed in the function doc as a precondition, rather than as an inline test.

Reply:
Could you recommend a better cap, or should I use no cap at all? I didn't want the computation to get too expensive but wanted to provide as much functionality as would be needed.

Reply:
To the extent that we're doing this function (and as I mentioned above, I have concerns), then it should allow users to input any integer number of Trotter steps. If they ask for 10 million Trotter steps and run out of memory its their own fault! As a minor comment, the number of trotter steps is called the Trotter number, which would perhaps be a better name.

Reply:
Haha alright I changed the name, and l'll leave the memory to the user!